### ISSUE TITLE
No post logout redirect ?

### ISSUE THREAD

**Issue by SebiNUI** [ThunderDev1/reactjs-ts-identityserver#1]
@ThunderDev1 I'm trying to use the post logout redirect within a similar SPA client... I noticed you didn't follow the same approach. My question is: did you manage to make post logout redirect work ?

**Comment by ThunderDev1** [comment]
Unfortunately no, I didn't get it to work. I did research the issue and as you probably saw, other people seem to have problems making it work.. I'll try to look in to it again when I have the time

**Comment by SebiNUI** [comment]
@ThunderDev1 I did, by creating the following method in the account controller:

[HttpGet]
        [AllowAnonymous]
        public async Task<IActionResult> Logout(string logoutId)
        {
            var logout = await _interaction.GetLogoutContextAsync(logoutId);
            await _signInManager.SignOutAsync();
            _logger.LogInformation("User logged out.");
            return Redirect(logout.PostLogoutRedirectUri);
        }

**Comment by shamilaprabhath** [comment]
Could you please share a similar project using JSX instead of TypeScript?

**Comment by ThunderDev1** [comment]
For any new questions, please start a new thread

**Comment by floodi** [comment]
I think id_token_hint should be used in logout-function(Spa - Nav.tsx):
  const logout = (event: any) => {
       event.preventDefault();
        **userManager.getUser()
        .then(user => {
          userManager.signoutRedirect({ 'id_token_hint': user.id_token });
        });**
        userManager.removeUser();
    };

**Comment by ThunderDev1** [comment]
Yes, that seems to be it, according to [http://docs.identityserver.io/en/latest/endpoints/endsession.html](this) you also need to pass the post_logout_redirect_uri. For some reason identity server is not providing the id_token, not sure what is missing.
